Lib/test/test_functools.py

```diff
@@ -362,14 +364,19 @@ def test_setstate(self):
         f = self.partial(signature)
         f.__setstate__((capture, (PH, 1), dict(a=10), dict(attr=[])))
         self.assertEqual(signature(f), (capture, (PH, 1), dict(a=10), dict(attr=[])))
-        with self.assertRaises(TypeError):
+        with self.assertRaises(TypeError) as cm:
             self.assertEqual(f(), (PH, 1), dict(a=10))
+        expected = ("missing positional arguments in 'partial' call; "
+                    "expected at least 1, got 0")
+        self.assertEqual(cm.exception.args[0], expected)
```

Comment:
`args[0]` is rather an implementation detail. We are more interesting in the string representation of the error message.
```suggestion
        self.assertEqual(str(cm.exception), expected)
```
zzzzzyou can also use `.assertRaisesRegexp()`, but with long error message it may be not better.